python/dgl/transforms/functional.py

```diff
@@ -1831,17 +1842,49 @@ def add_self_loop(g, etype=None):
     >>> g = dgl.add_self_loop(g, etype='follows')
     >>> g
     Graph(num_nodes={'user': 3, 'game': 2},
-        num_edges={('user', 'plays', 'game'): 2, ('user', 'follows', 'user'): 5},
-        metagraph=[('user', 'user'), ('user', 'game')])
+          num_edges={('user', 'plays', 'game'): 2, ('user', 'follows', 'user'): 5},
+          metagraph=[('user', 'user'), ('user', 'game')])
     """
     etype = g.to_canonical_etype(etype)
+    data = {}
+    reduce_funcs = {'sum': function.sum,
+                    'mean': function.mean,
+                    'max': function.max,
+                    'min': function.min}
+
+    if edge_feat_names is None:
+        edge_feat_names = g.edges[etype].data.keys()
+
     if etype[0] != etype[2]:
         raise DGLError(
             'add_self_loop does not support unidirectional bipartite graphs: {}.' \
             'Please make sure the types of head node and tail node are identical.' \
             ''.format(etype))
+
+    for feat_name in edge_feat_names:
+        if isinstance(fill_data, (int, float)):
+            dtype = g.edges[etype].data[feat_name].dtype
+            tmp_fill_data = F.copy_to(F.tensor([fill_data], dtype), g.device)
+            try:
+                data[feat_name] = F.zeros(
+                    (g.num_nodes(etype[0]), g.edges[etype].data[feat_name].shape[1]), dtype,
+                    g.device) + tmp_fill_data
+            except IndexError:
```

Comment:
Rather than use the try-except logic. You can first check the shape of `g.edges[etype].data[feat_name]`.